recipes/usockets/all/conanfile.py

```diff
@@ -0,0 +1,128 @@
+import os
+
+from conans import ConanFile, tools, MSBuild
+from conans.errors import ConanInvalidConfiguration
+
+
+class UsocketsConan(ConanFile):
+    name = "usockets"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Miniscule cross-platform eventing, networking & crypto for async applications"
+    license = "Apache-2.0"
+    homepage = "https://github.com/uNetworking/uSockets"
+    topics = ("conan", "socket", "network", "web")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"fPIC": [True, False],
+               "with_ssl": [False, "openssl", "wolfssl"],
+               "with_libuv": [True, False]}
+    default_options = {"fPIC": True, "with_ssl": False, "with_libuv": True}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+
+    def requirements(self):
+        if self.options.with_ssl == "openssl":
+            self.requires("openssl/1.1.1g")
+        elif self.options.with_ssl == "wolfssl":
+            self.requires("wolfssl/4.4.0")
+        if self.options.with_libuv:
+            self.requires("libuv/1.38.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("uSockets-%s" % self.version, self._source_subfolder)
+
+    def _build_msvc(self):
+        with tools.chdir(os.path.join(self._source_subfolder)):
+            tools.replace_in_file("uSockets.vcxproj",
```

Comment:
Maybe all the `tools.replace_in_file` could be replaced with one patch?

Reply:
Added a pair of patches for makefile (based on @madebr proposal) and vcxproj. Looks cleaner now.